### PR TITLE
chore(Python): create heap-sort

### PR DESCRIPTION
**Pull request by Sranu2109**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Other information:**

Created heap sort in python.

### CODE REVIEW

**Comment by sridhar-5** [comment]
I'm checking it now.

File: algorithms/Python/sorting/heap-sort.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        largest = index
+    if (r < size and alist[r] > alist[largest]):
+        largest = r
+    if (largest != index):
+        alist[largest], alist[index] = alist[index], alist[largest]
+        max_heapify(alist, largest, size)
+ 
+ 
+alist = input('Enter the list of numbers: ').split()
+alist = [int(x) for x in alist]
```
Comment (sridhar-5) [inline algorithms/Python/sorting/heap-sort.py:40]: There is a failing test that is pointing to this line. I think replacing the line 40 with the snippet below might solve this.

if you still have a failing test please comment.

alist = [x for x in alist]
alist = list(map(int, alist))

**Review by sridhar-5 — CHANGES_REQUESTED** [review]
check the comments I've mentioned and do the changes accordingly. if you have trouble again. please comment I would be happy to help here.

**Comment by sridhar-5** [comment]
Not really sure why test fails all the time. But personally this is the efficient way of doing this the test suggests in using `str` in place of `int`. But only `Int` has to be used. so the code looks good to me other than that. I personally feel this code is right and works.

Thanks and great job @Sranu2109

**Review by sridhar-5 — APPROVED** [review]
Code looks good to me. other than that failing test. if it's not a problem you can take it @ming-tsai

**Review by ming-tsai — CHANGES_REQUESTED** [review]
Hi @Sranu2109,
Please add your code on the readme of Python

**Comment by sridhar-5** [comment]
@ming-tsai ,I think the read me is now updated by @Sranu2109. can you check ?

**Review by ming-tsai — APPROVED** [review]

**Comment by ming-tsai** [comment]
> @ming-tsai ,I think the read me is now updated by @Sranu2109. can you check ? 

Thanks

**Review by ming-tsai — CHANGES_REQUESTED** [review]
Hi @Sranu2109,
Could you check why Python GitHub action is failed?

**Comment by Sranu2109** [comment]
> Hi @Sranu2109,
> Could you check why Python GitHub action is failed?

You could use Pip-Installer from Github Actions to run package installation scripts, refer to this article [here](https://github.com/marketplace/actions/pip-installer).

It seems that the `mypy` package is missing from the build.

And/or you could add a `requirements.txt` top your project root and modify your `actions.yaml` to this.
```
steps:
- uses: actions/checkout@v2
- name: Set up Python
  uses: actions/setup-python@v2
  with:
    python-version: '3.x'
- name: Install dependencies
  run: |
    python -m pip install --upgrade pip
    pip install -r requirements.txt 
```

**Comment by sridhar-5** [comment]
can you add this import statement and update here if it works?.Not sure though just give it a try.

`from typing import Callable`

File: algorithms/Python/sorting/heap-sort.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    if (r < size and alist[r] > alist[largest]):
+        largest = r
+    if (largest != index):
+        alist[largest], alist[index] = alist[index], alist[largest]
+        max_heapify(alist, largest, size)
+
+
+alist = input('Enter the list of numbers: ').split()
+alist = [x for x in alist] 
+alist = list(map(int, alist))
```
Comment (sridhar-5) [inline algorithms/Python/sorting/heap-sort.py:41]: And also try typecasting alist to int.

**Review by sridhar-5 — CHANGES_REQUESTED** [review]

**Comment by sridhar-5** [comment]
If that didn't work. update here I'll find other ways.

**Review by Arsenic-ATG — COMMENTED** [review]
the python check was failing due to the bug in mypy where it fails to identify callable types. 

it is also reported as a bug on their repo, you can check that [here](https://github.com/python/mypy/issues/8117)

**Review by sridhar-5 — APPROVED** [review]
Nice Work @Sranu2109  This can be Merged @ming-tsai.

Thanks.

**Comment by Arsenic-ATG** [comment]
the last 2 commits I did were minute changes which required to pass test cases, and the changes include :- 

1. Adding extra newline between import and rest of the code ( to pass isort check )

2. Adding automated test cases instead of using input() and thus removing the interactive element from the program making unit testing possible. ( to pass pytest check )

**Review by ming-tsai — APPROVED** [review]

**Comment by sridhar-5** [comment]
Great @Arsenic-ATG.

**Comment by ming-tsai** [comment]
Nice team work

**Pull request merged**